components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java

```diff
@@ -212,424 +171,357 @@ public Map<String, Integer> getActionsCountPerType(Integer tenantId) throws Acti
+        try {
+            // Delete existing properties.
+            deleteActionPropertiesInDB(updatingActionDTO.getId(),
+                    new ArrayList<>(existingActionDTO.getProperties().keySet()), tenantId);
+            // Add updated properties.
+            addActionPropertiesToDB(updatingActionDTO.getId(), updatingProperties, tenantId);
+        } catch (TransactionException e) {
+            throw new ActionMgtServerException("Error while updating Action Properties.", e);
+        }
     }
 
     /**
-     * Update Action Endpoint properties.
+     * Add Action properties to the Database.
      *
-     * @param actionId           UUID of the created Action.
-     * @param endpointProperties Endpoint Properties to be updated.
-     * @param tenantId           Tenant ID.
+     * @param actionId         UUID of the created Action.
+     * @param actionProperties Properties of the Action.
+     * @param tenantId         Tenant ID.
+     * @throws TransactionException If an error occurs while persisting action properties to the database.
      */
-    private void updateActionEndpointProperties(String actionId, Map<String, String> endpointProperties,
-                                                Integer tenantId) throws ActionMgtException {
+    private void addActionPropertiesToDB(String actionId, Map<String, String> actionProperties, Integer tenantId)
+            throws TransactionException {
 
         NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
-        try {
-            jdbcTemplate.withTransaction(template -> {
-                template.executeBatchInsert(ActionMgtSQLConstants.Query.UPDATE_ACTION_ENDPOINT_PROPERTIES,
+        jdbcTemplate.withTransaction(template -> {
+            template.executeBatchInsert(ActionMgtSQLConstants.Query.ADD_ACTION_ENDPOINT_PROPERTIES,
                     statement -> {
-                        for (Map.Entry<String, String> property : endpointProperties.entrySet()) {
-                            statement.setString(ActionMgtSQLConstants.Column.ACTION_ENDPOINT_PROPERTY_VALUE,
-                                    property.getValue());
-                            statement.setString(ActionMgtSQLConstants.Column.ACTION_ENDPOINT_PROPERTY_NAME,
-                                    property.getKey());
+                        for (Map.Entry<String, String> property : actionProperties.entrySet()) {
                             statement.setString(ActionMgtSQLConstants.Column.ACTION_ENDPOINT_UUID, actionId);
                             statement.setInt(ActionMgtSQLConstants.Column.TENANT_ID, tenantId);
+                            statement.setString(ActionMgtSQLConstants.Column.ACTION_ENDPOINT_PROPERTY_NAME,
+                                    property.getKey());
+                            statement.setString(ActionMgtSQLConstants.Column.ACTION_ENDPOINT_PROPERTY_VALUE,
+                                    property.getValue());
```

Comment:
Let's rename the column name constants as this is now related to properties not just endpoint properties

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6119/commits/dab9dd75643167cc04aa5d35aec2f044d59a58d4